### PR TITLE
fix(outputs.execd): Fixing the execd behavior to not throw error when partially unserializable metrics are written.

### LINKED ISSUE #11727: Specific prometheus metrics involving NaN causes serialization error, throwing away all other metrics
### Relevant telegraf.conf

```toml
[[inputs.prometheus]]
  tagexclude = ["url"]

  ignore_timestamp = true
  metric_version = 2

[[outputs.execd]]
  command = ["java", "-Xms64m", "-Xmx128m", "-jar", "/opt/metrics_exporter/lib/metrics_exporter.jar"]
  data_format = "influx"
```

### Logs from Telegraf

```text
2022-08-26T09:28:25Z D! [agent] Starting service inputs
2022-08-26T09:29:09Z D! [serializers.influx] could not serialize field "latency_request": is NaN; discarding field
2022-08-26T09:29:09Z D! [serializers.influx] could not serialize field "latency_request": is NaN; discarding field
2022-08-26T09:29:09Z D! [serializers.influx] could not serialize field "latency_request": is NaN; discarding field
2022-08-26T09:29:26Z D! [serializers.influx] could not serialize field "latency_request": is NaN; discarding field
2022-08-26T09:29:26Z D! [outputs.execd] Buffer fullness: 129 / 10000 metrics
2022-08-26T09:29:26Z E! [agent] Error writing to outputs.execd: error serializing metrics: "prometheus,host=internal-host,quantile=0.5,service=web-service": no serializable fields
```

### System info

Telegraf 1.22.0, CentOS Linux 7

### Docker

_No response_

### Steps to reproduce

1. Use prometheus input plugin with `metric_version = 2` configured to load the prometheus metrics output including the following:
```
# HELP latency_request Metric autogenerated by statsd_exporter.
# TYPE latency_request summary
latency_request{service="web-service",quantile="0.5"} NaN
latency_request{service="web-service",quantile="0.9"} NaN
latency_request{service="web-service",quantile="0.99"} NaN
latency_request_sum{service="web-service"} 0
latency_request_count{service="web-service"} 1
```
2. Make sure that latency_request field is the only field with the combination of the following tag `{service="web-service",quantile="0.5"}`
4. Allow execd outputs plugin to load the data as influx data format
5. As a result, the influx data has a line containing metrics without field and measurement, causing an error and throwing away all other metrics that we wish to collect.
...

### Expected behavior

When serializer discards a field, it should check if it would leave a metrics without field and measurement. If so, it should drop the entire metrics.
```
[serializers.influx] could not serialize field "latency_request": is NaN; discarding field
[agent] Error writing to outputs.execd: error serializing metrics: "prometheus,host=internal-host,quantile=0.5,service=web-service": no serializable fields
```

### Actual behavior

When serializer discards a field, it does not check if it would leave a metrics without field and measurement. And when loading the data to execd output plugin, it finds a metrics with no serialized fields, causing all other metrics to be dropped
```
[serializers.influx] could not serialize field "latency_request": is NaN; discarding field
[agent] Error writing to outputs.execd: error serializing metrics: "prometheus,host=internal-host,quantile=0.5,service=web-service": no serializable fields
```

### Additional info

Names of metrics, tags, filenames are obfuscated due to a security concern, However, the obfuscation of each files are consistent, such that any name that appears multiple times did so in the original files as well.

### PR DESCRIPTION
**Pull request by lsegawa**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11727

When partially unserializable metrics are written to execd, it used to throw error and drop all other metrics. This change will allow partially unserializable metric to log a warning message and continue to next metric.

### CODE REVIEW

**Review by MyaLongmire — APPROVED** [review]
Thanks for this pr!

File: plugins/outputs/execd/execd.go [hunk 1]
```diff
@@ -81,7 +81,8 @@ func (e *Execd) Write(metrics []telegraf.Metric) error {
 	for _, m := range metrics {
 		b, err := e.serializer.Serialize(m)
 		if err != nil {
-			return fmt.Errorf("error serializing metrics: %s", err)
+			e.Log.Warn("Skipping metric due to a serialization error: %s", err)
```
Comment (Hipska) [inline plugins/outputs/execd/execd.go:84]: This decreases the log message severity from Error to Warning. Not sure if this is good.

Anyway, please use `%w` for errors.

**Review by Hipska — CHANGES_REQUESTED** [review]
I would expect this to be a configurable setting, so by default users have the same behaviour as before, and they can choose to ignore incomplete metrics.

However, after typing this, I don't think this is the place to "fix" this.

File: plugins/outputs/execd/execd.go (hunk 1 above)
Comment (srebhan) [reply plugins/outputs/execd/execd.go:84]: Using `%w` (i.e. error wrapping) does _not_ make sense for logging as we will never unwrap the error. It only makes sense when returning an error and allows us to test for the underlying (wrapped) error type.

**Comment by srebhan** [comment]
@lsegawa thanks for your contribution, I think it is very valuable!

However, I tend to agree with @Hipska that this change should be guarded with a config option e.g. `ignore_serialization_errors` set to `false` to reproduce the current behavior. What do you think?

Furthermore, please sign the [Contributor license agreement (CLA)](https://www.influxdata.com/legal/cla/) as otherwise we cannot merge your PR.

File: plugins/outputs/execd/execd.go (hunk 1 above)
Comment (Hipska) [reply plugins/outputs/execd/execd.go:84]: It is more clear what is expected as operand, it seems a good practice to always use `%w` when dealing with errors, just as you use `%d` for integers or `%t` for booleans.

**Comment by lsegawa** [comment]
!signed-cla

**Comment by lsegawa** [comment]
Hi everyone,

Thank you very much for you feedback. Please allow me some time to address them.

Thanks!

File: plugins/outputs/execd/sample.conf [hunk 2]
```diff
@@ -13,6 +13,11 @@
   ## Delay before the process is restarted after an unexpected termination
   restart_delay = "10s"
 
+  ## Flag to determine whether execd should throw error when part of metrics is unserializable
+  ## Setting this to true will skip the unserializable metrics and process the rest of metrics
+  ## Setting this to false will throw error when encountering unserializable metrics and none will be processed
+  ignore_serialization_error = false
```
Comment (Hipska) [inline plugins/outputs/execd/sample.conf:19]: Defaults can be commented:
```suggestion
  # ignore_serialization_error = false
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: plugins/outputs/execd/sample.conf (hunk 2 above)
Comment (lsegawa) [reply plugins/outputs/execd/sample.conf]: I believe I fixed it!

File: plugins/outputs/execd/execd.go (hunk 1 above)
Comment (lsegawa) [reply plugins/outputs/execd/execd.go]: I believe I fixed it!

**Review by Hipska — APPROVED** [review]

**Pull request merged**